src/watch_filter.rs

```diff
@@ -76,19 +86,24 @@ impl WatchFilter {
         })
     }
 
-    fn matches_one_of_ignore_patterns(&self, event: &Event) -> bool {
+    fn matches_one_of_watched_patterns(&self, event: &Event) -> bool {
         event.paths().any(|(path, _)| {
-            self.ignore_patterns
-                .iter()
-                .any(|pattern| pattern.matches_path(path))
+            self.config.manifest_pattern.pattern.matches_path(path)
+                || self
+                    .config
+                    .source_patterns
+                    .iter()
+                    .chain(self.config.artifact_patterns.iter())
+                    .any(|wp| wp.active && wp.pattern.matches_path(path))
         })
     }
 
-    fn matches_one_of_path_patterns(&self, event: &Event) -> bool {
+    fn matches_one_of_ignore_patterns(&self, event: &Event) -> bool {
         event.paths().any(|(path, _)| {
-            self.path_patterns
+            self.config
+                .ignore_patterns
                 .iter()
-                .any(|pattern| pattern.matches_path(path))
+                .any(|wp| wp.active && wp.pattern.matches_path(path))
```

Comment:
I like how you subtly changed the name of the lambda argument to avoid `cargo fmt` messing up the code structure.  _shakes puny fist at `cargo fmt`_